### PR TITLE
Commit Message

### PR DESCRIPTION
**Pull request by allohal**

### CODE REVIEW

File: VectTemplate.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+// Created by alloha_l on 15.07.18.
+//
+
+#ifndef UNTITLED_VECTTEMPLATE_H
+#define UNTITLED_VECTTEMPLATE_H
+
+#include <iostream>
+#include <vector>
+
+template<typename T> void func(std::vector<T> t, std::string name){
```
Comment (Morheit) [inline VectTemplate.h]: Why `std::vector<T>` is passed by value? Pass it by reference `&`. The same for `std::string`

File: VectTemplate.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    std::cout << "\nType: " << name << std::endl;
+    std::cout<<"Number of members before deleting "<< t.size()<<std::endl;
+
+    if(!t.empty()){
+        n = t.back();
+        t.pop_back();
+    }
+
+    std::cout<<"Number of members after deleting "<< t.size()<<std::endl;
+    for(T i: t){
```
Comment (Morheit) [inline VectTemplate.h]: Use `const &` in range-based loop. Example: `const T& i : t`

File: VectTemplate.h [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        std::cout<<i<<'\n';
+    }
+
+    std::cout<<"Number of members before adding "<< t.size()<<std::endl;
+    t.push_back(n);
+
+    for(T i: t){
```
Comment (Morheit) [inline VectTemplate.h]: Use `const &` in range-based loop. Example: `const T& i : t`

File: VectTemplate.h [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+#include <iostream>
+#include <vector>
+
+template<typename T> void func(std::vector<T> t, std::string name){
+    T n;
+    std::cout << "\nType: " << name << std::endl;
+    std::cout<<"Number of members before deleting "<< t.size()<<std::endl;
+
+    if(!t.empty()){
```
Comment (Morheit) [inline VectTemplate.h]: That's would be nice if you add some spacing between `if` and `(`. The same for `)` and `{`

File: VectTemplate.h [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        n = t.back();
+        t.pop_back();
+    }
+
+    std::cout<<"Number of members after deleting "<< t.size()<<std::endl;
```
Comment (Morheit) [inline VectTemplate.h]: Please, add some space between `code` and `<<`

Comment (allohal) [reply VectTemplate.h]: done

File: VectTemplate.h (hunk 4 above)
Comment (allohal) [reply VectTemplate.h]: done

File: VectTemplate.h (hunk 3 above)
Comment (allohal) [reply VectTemplate.h]: done

File: VectTemplate.h (hunk 2 above)
Comment (allohal) [reply VectTemplate.h]: done

**Pull request closed without merging**